inference-engine/.gitignore

```diff
@@ -0,0 +1,30 @@
+# vim temporary files
```

Comment:
we already have .gitignore in the root, we can merge this one there

Reply:
`.gitignore` is designed to be able to add such a file in any subdirectory within a repository, so I didn't see an issue with keeping it here. 

But I don't have a strong preference, so I'm happy to merge them into the root-level `.gitignore` file.